sdks/python/apache_beam/dataframe/schemas_test.py

```diff
@@ -203,19 +223,32 @@ def test_batch_with_df_transform(self):
                 proxy=schemas.generate_proxy(Animal)))
         assert_that(res, equal_to([('Falcon', 375.), ('Parrot', 25.)]))
 
+  def assert_typehints_equal(self, left, right):
+    left = typehints.normalize(left)
+    right = typehints.normalize(right)
+
+    if _match_is_named_tuple(left):
+      self.assertTrue(_match_is_named_tuple(right))
+      self.assertEqual(left.__annotations__, right.__annotations__)
+    else:
+      self.assertEqual(left, right)
+
   @parameterized.expand(SERIES_TESTS + NOINDEX_DF_TESTS)
-  def test_unbatch_no_index(self, df_or_series, rows):
+  def test_unbatch_no_index(self, df_or_series, rows, beam_type):
     proxy = df_or_series[:0]
 
     with TestPipeline() as p:
       res = (
           p | beam.Create([df_or_series[::2], df_or_series[1::2]])
           | schemas.UnbatchPandas(proxy))
 
+      # Verify that the unbatched PCollection has the expected typehint
+      self.assert_typehints_equal(res.element_type, beam_type)
```

Comment:
Does `self.assertTrue(typehints.is_consistent_with(res.element_type, beam_type))` work here? Do you need assert_typehints_equal because typing.NamedTuples get converted to Any? If so, could you reference https://issues.apache.org/jira/browse/BEAM-8538 here?

Reply:
I don't think BEAM-8583 is the issue, the problem seems to be just that `sub` and `base` are two distinct NamedTuple types. They have the same fields in the same order, but that doesn't satisfy the `issubclass` check. https://github.com/apache/beam/blob/fd33f16882fcedd77ec1e4ab553328bce4a277ac/sdks/python/apache_beam/typehints/typehints.py#L1179

Perhaps I should add logic for checking the types of NamedTuple fields to `is_consistent_with`, something like:
```
      if (_match_is_named_tuple(base))
        # Check if any of sub's parents are NamedTuples?
        self.assertEqual(sub.__annotations__, base.__annotations__)
 ```

Reply:
There shouldn't be `typing` module types during type checking. They should get converted to internal types or Any.

The solution would be to write an internal type for typing.NamedTuple, and use that type internally.

Reply:
I see. I suppose we should be able to map them to RowTypeConstraint. I'll add a TODO(BEAM-8538) and make `_match_is_named_tuple` public

Reply:
Done.